solr/core/src/java/org/apache/solr/client/solrj/embedded/EmbeddedSolrServer.java

```diff
@@ -215,8 +215,9 @@ public NamedList<Object> request(SolrRequest<?> request, String coreName)
       if (handler == null) {
         throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "unknown handler: " + path);
       }
-
-      req = _parser.buildRequestFrom(core, params, getContentStreams(request));
+      req =
+          _parser.buildRequestFrom(
```

Comment:
I wonder, should we have still have a _parser.buildRequestFrom() that doesn't require the user principal?   These days, do we need to make sure that every method has a user principal?   Maybe what I am asking is, are there lots of other places where, if you enable basic auth, then you get errors because the oroginal code didn't plan for that?

Reply:
We could remove the buildRequestFrom() that doesn't take a Principal; outside of tests, it's used in only two places. Where it's used in `EmbeddedSolrServer.request(SolrRequest<?> request, String coreName)`, we could get the Principal from the SolrRequest. Where it's used in `DirectSolrConnection.request(SolrRequestHandler handler, SolrParams params, String body)` I think we'd need to pass a null Principal as I'm not seeing where to get one from in that context.

About the larger question -- are there lots of other cases where enabling basic auth causes errors -- I don't think I know enough to give a good answer. I do think that the subquery use case is the most obvious/glaring place for such an error to arise because it involves creating and executing a brand new query.